app/src/main/java/com/example/chattutorial/ChannelActivity3.kt

```diff
@@ -44,27 +46,26 @@ class ChannelActivity3 : AppCompatActivity(R.layout.activity_channel_3) {
             }
 
         val channelController = ChatClient.instance().channel(cid)
-        val currentlyTyping = MutableLiveData<List<String>>(ArrayList())
+        val currentlyTyping = MutableLiveData<Set<String>>(HashSet())
 
         channelController.events().subscribe {
-            val name = it.user!!.extraData["name"]!! as String
-            val typing = currentlyTyping.value ?: listOf()
-            val typingCopy: MutableList<String> = typing.toMutableList()
+            val typing = currentlyTyping.value ?: setOf()
+            val typingCopy: MutableSet<String> = typing.toMutableSet()
             when (it) {
                 is TypingStartEvent -> {
-                    if (typingCopy.contains(name).not()) {
-                        typingCopy.add(name)
-                    }
+                    val name = it.user.extraData["name"]!! as String
```

Comment:
The `as String` cast here will make sure that this is a non-null value (and fail with a NPE if it is), so the separate `!!` is not required